bridge/README.md

```diff
@@ -0,0 +1,71 @@
+
+There is no change to the circulating supply of your token when it crosses the bridge:
+- Tokens that leave the Ethereum network are locked and the same number of tokens are minted on Polygon PoS as a pegged token (1:1).
+- To move the tokens back to the Ethereum network, tokens are burned on Polygon PoS network and unlocked on Ethereum network.
+
+## Listner
+
+The bridge module has a `BaseListner` which is extended by `RootChainListener`, `MaticChainListener` and `HeimdallListener`. It has all the methods required to start polling, listen to incoming headers(blocks) and stop the process if required. All the 3 listneres uses these properties with their individual implementations on how to handle the incoming header once received.
+
+For example in `RootChainListner` the incoming header is used to determine the current height of root chain and calculate the `from` and `to` block numbers using which the events are fetched from the root chain. These events are then sent to `handleLog` where based on their event signature they are added to queue as tasks for further processing by their respective processors.
+
+## Processor
+
+There is a `BaseProcessor` which is extended by every processor in the processor package. It has all the methods required to start processing the events, Register for the tasks which are there in the queue and stop the process if required. All the processors uses these properties with their individual implementations on how to handle the incoming events once received.
+
+Once the event is added to the queue by the `Listner`, The `Processor` takes over and processes the events which are added into the queue based on their event signature. Each processor has `RegisterTasks` method using which they register to process specific tasks added to the queue based on the module they serve, For example `StakingProcessor` registers for `Staking` related tasks, `ClerkProcessor` registers for `Clerk` related tasks and so on. You can look into each processor to check which tasks they are registered for.
+
+## How to start bridge
+
+Bridge should used by validator nodes only as they are the only ones who can send txns on heimdall chain, So if a non validator node is running bridge then it's of no use as it won't be able to send txns to heimdall chain.
+
+To start bridge as a validator you just have to add `--bridge` command to your heimdall node command, You can also control which services you want to run from processor by using `--all` or `--only` flags.
+
+To run all the service :
+
+```bash
+./heimdalld start --bridge --all
+```
+
+To use specific services :
+
+```bash
+./heimdalld start --bridge --only=clerk,staking
+```
+
+## Reset
+
+> :warning: Do this only when you are advised so and you understand the impact of this command. 
+
+If you want to reset the bridge server data, you can use the following command:
+
+```bash
+./heimdalld unsafe-reset-all
+```
+
+## Common Issues (FAQ)
+
```

Comment:
**1. Connection reset by peer**

If the node shows up the logs in below pattern 

panic: Exception (501) Reason: "read tcp 127.0.0.1:35218->127.0.0.1:5672: read: connection reset by peer"
goroutine 1 [running]:
github.com/maticnetwork/heimdall/bridge/setu/queue. NewQueueConnector({0xc0002ccf60,
0x22})
+0x26c

Your Heimdall Bridge has issues please follow the steps below to fix this

stop rabbitmq-server
rm /var/lib/rabbitmq/mnesia
start rabbitmq-server

**2.Validator is unable to propose checkpoints**

Please check the validator heimdall.service and ensure the below flag is set and ensure rabbitmq is installed on your validator.

Once done restart the services.

 **--bridge --all**


3.If your node shows up the below log snips 

**ERROR[2023-02-13|22:47:32.081] Nonce for the given event not in order       module=staking service=processor validatorId=1** 

If the following logs are found on a large frequency on Heimdall then this issue is related to the bridge service resetting the bridge directory fixes this issue.

sudo service rabbitmq-server stop
mv /var/lib/rabbitmq/mnesia /var/lib/rabbitmq/mnesia-old
sudo service rabbitmq-server start